sdk/platform/http_client/curl/src/az_curl.c

```diff
@@ -478,18 +476,22 @@ _az_http_client_curl_setup_url(CURL* p_curl, _az_http_request const* p_request)
   AZ_PRECONDITION_NOT_NULL(p_curl);
   AZ_PRECONDITION_NOT_NULL(p_request);
 
+  az_span request_url = { 0 };
+  AZ_RETURN_IF_FAILED(az_http_request_get_parts(p_request, NULL, &request_url, NULL));
+  int32_t request_url_size = az_span_size(request_url);
+
   az_span writable_buffer;
   {
     // Add 1 for 0-terminated str
-    int32_t const url_final_size = p_request->_internal.url_length + 1;
+    int32_t const url_final_size = request_url_size + 1;
 
     // allocate buffer to add \0
     AZ_RETURN_IF_FAILED(_az_span_malloc(url_final_size, &writable_buffer));
   }
 
   // write url in buffer (will add \0 at the end)
   az_result result = _az_http_client_curl_append_url(
-      writable_buffer, az_span_slice(p_request->_internal.url, 0, p_request->_internal.url_length));
+      writable_buffer, az_span_slice(request_url, 0, request_url_size));
```

Comment:
Is this slice and `request_url_size` local necessary? Isn't `request_url` already the size of `request_url_size`?

Reply:
nice catch!